lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java

```diff
@@ -1224,10 +1259,29 @@ public int compare(BytesRef o1, BytesRef o2) {
     }
   }
 
-  private int addStemException(String stemException) {
-    stemExceptions = ArrayUtil.grow(stemExceptions, stemExceptionCount + 1);
-    stemExceptions[stemExceptionCount++] = stemException;
-    return stemExceptionCount; // we use '0' to indicate no exception for the form
+  private List<String> readMorphFields(String word, String unparsed) {
+    List<String> morphFields = null;
+    for (String datum : splitMorphData(unparsed)) {
+      if (datum.startsWith("ph:")) {
+        addPhoneticRepEntries(word, datum.substring(3));
+      } else {
+        if (morphFields == null) morphFields = new ArrayList<>(1);
+        morphFields.add(datum);
+      }
+    }
+    return morphFields == null ? Collections.emptyList() : morphFields;
+  }
+
+  private int addMorphFields(Map<String, Integer> indices, String morphFields) {
+    Integer alreadyCached = indices.get(morphFields);
```

Comment:
indices.computeIfAbsent?

Reply:
Thanks, but my sense of beauty prefers `computeIfAbsent` lambdas to be side-effect free

Reply:
There is absolutely nothing stating they have to be though?... it's just a contract on updating the map - how it computes the value and what it updates along the way is not part of the contract at all.

I'll leave it to your sense of beauty though, of course.

Reply:
Yes, I've checked the javadoc and found nothing as well. But still, it aligns with other STM-like similar APIs (e.g. `AtomicInteger#getAndAccumulate`) and just feels better :)